modulesync 4.0.0

bastelfreak opened the pull request:
modulesync 4.0.0

bastelfreak committed:
modulesync 4.0.0

ekohl commented on .rubocop.yml:
```diff
@@ -1,4 +1,3 @@
 ---
-inherit_from: '.rubocop_todo.yml'
```
If this is removed, the todo file itself should also be removed.

ekohl reviewed (changes requested):
No Puppet requirement found

ekohl committed:
Declare Puppet version compatibility

This allows our testing matrix to verify the support. This is a good
practice. Puppet may change its API's and then you have no idea if it
actually works.

ekohl reviewed (approved):
:+1: if it passes, but I also made some changes. Wondering if that deserves its own PR for the changelog.

bastelfreak commented:
created https://github.com/voxpupuli/puppet-xmlfile/issues/15 which mentions the puppet version range
